cg/meta/upload/gisaid/gisaid.py

```diff
@@ -293,33 +345,42 @@ def get_accession_numbers(self, case_id: str) -> dict[str, str]:
                     )
                 else:
                     continue
-                accession_obj = GisaidAccession(log_message=log_message)
-                accession_numbers[accession_obj.sample_id] = accession_obj.accession_nr
+                gisaid_accession = GisaidAccession(log_message=log_message)
+                accession_numbers[gisaid_accession.sample_id] = gisaid_accession.accession_nr
         return accession_numbers
 
-    def update_completion_file(self, case_id: str) -> None:
-        """Update completion file with accession numbers"""
-        completion_file = self.get_completion_file_from_hk(case_id=case_id)
-        accession_dict = self.get_accession_numbers(case_id=case_id)
-        completion_df = self.get_completion_dataframe(completion_file=completion_file)
-        completion_df["GISAID_accession"] = completion_df["provnummer"].apply(
-            lambda x: accession_dict[x]
+    def update_complementary_file_with_gisaid_accessions(self, case_id: str) -> None:
+        """Update complementary file with GISAID accession numbers."""
+        complementary_report: File | None = self.get_complementary_file_from_hk(case_id=case_id)
+        accession: dict = self.get_gisaid_accession_numbers(case_id=case_id)
+        sars_cov_complementary_reports: list[GisaidComplementaryReport] = (
+            self.parse_and_get_sars_cov_complementary_reports(Path(complementary_report.full_path))
         )
-        completion_df.to_csv(
-            completion_file.full_path,
-            sep=",",
-            index=False,
+        self.add_gisaid_accession_to_complementary_reports(
+            gisaid_accession=accession, reports=sars_cov_complementary_reports
+        )
+        reports: list[dict] = [report.model_dump() for report in sars_cov_complementary_reports]
+        write_csv_from_dict(
+            content=reports, fieldnames=HEADERS, file_path=Path(complementary_report.full_path)
```

Comment:
What headers are required here and does it does it have to be in Swedish? @Clinical-Genomics/prodbioinfo

Reply:
I would keep the headers as is, this format was not defined by us iirc